libopae++/include/opaec++/pvalue.h

```diff
@@ -104,9 +106,13 @@ struct guid_t {
     return ostr;
   }
 
+  bool is_set() const { return is_set_; }
+  void is_set(bool b) { is_set_ = b;    }
```

Comment:
Shouldn't this be private and only set to `true` when `fpgaProperties(Set|Get).*` returns FPGA_OK?
Would it be better if we called this `invalidate()` that didn't take any arguments? All it would do is set `is_set_` to `false`.

Reply:
Sure, and we'd need to make properties a friend of pvalue..

Reply:
why? properties shouldn't need to set `is_set_` to `true` - should it? all it needs to do is call `p.invalidate()` to set it to false... no?

Reply:
Yeah, but if invalidate() is to be private in pvalue, then properties will need to be a friend. Refer to the copy assignment operator.

Reply:
why can't `invalidate` be public? I'm ok with making properties a `friend` too. It makes sense they should be friends 😸